### DISCUSSION TITLE
Select and list the same parameter from multiple models

### DISCUSSION THREAD

**Discussion by JulianGaviriaL** [easystats/parameters#927] [Q&A]
I have multiple (>300) models like:
```
m001 <- lm(Observations ~ Group ,data=r_001)
sm001<-standardize_parameters(m001)

> sm001
# Standardization method: refit

Parameter   | Std. Coef. |        95% CI
----------------------------------------
(Intercept) |       0.16 | [-0.41, 0.73]
Grouppre    |      -0.30 | [-0.90, 0.30]
GenderM     |      -0.18 | [-0.70, 0.34]
Age         |      -0.11 | [-0.35, 0.14]
EducationE2 |      -0.11 | [-0.78, 0.56]
EducationE3 |       0.13 | [-0.49, 0.74]

```

```
sm301<-standardize_parameters(m301)
> sm301
# Standardization method: refit

Parameter   | Std. Coef. |        95% CI
----------------------------------------
(Intercept) |       0.16 | [-0.41, 0.73]
Grouppre    |      0.38 | [-0.90, 0.60]
GenderM     |      -0.18 | [-0.70, 0.34]
Age         |      -0.11 | [-0.35, 0.14]
EducationE2 |      -0.11 | [-0.78, 0.56]
EducationE3 |       0.13 | [-0.49, 0.74]
```

**How can I select and list the same parameter (i.e, "Grouppre") from each "sm" object in the environment?** 

I unsucceed with the following loop:

```
require(foreach)
output_position <- grep("sm",ls())
all_effects <- foreach(i=output_position,.combine="rbind") %do% {
  output <- get(ls()[i])
  unlist(output["Grouppre",])
}
rownames(all_effects) <- ls()[output_position]
all_effects
```

**Comment by strengejacke** [comment]
The example looks strange, I wonder where the coefficients like `age` etc. come from. If you want to keep only specific parameters, use arguments `keep` and `drop` (see https://easystats.github.io/parameters/reference/model_parameters.default.html).

**Comment by mattansb** [comment]
``` r
m001 <- lm(mpg ~ ., data = mtcars)
m002 <- lm(hp ~ ., data = mtcars)

mod_names <- ls() |> Filter(f = \(x) startsWith(x, "m00"))
mod_list <- lapply(setNames(nm = mod_names), get)

parameters::compare_parameters(mod_list, 
                               keep = "am", 
                               standardize = "refit")
#> Parameter    |               m001 |               m002
#> ------------------------------------------------------
#> am           | 0.21 (-0.15, 0.56) | 0.07 (-0.25, 0.38)
#> ------------------------------------------------------
#> Observations |                 32 |                 32
```

<sup>Created on 2023-12-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by JulianGaviriaL** [reply]
Sorry, I posted the wrong model. and thanks for the tip about the keep option

```
m001 <- lm(Observations ~ Group+Gender+Age+Education ,data=r_001)
sm001<-standardize_parameters(m001,keep="Grouppre")

> sm001
# Standardization method: refit

Parameter   | Std. Coef. |        95% CI
----------------------------------------
Grouppre    |      -0.30 | [-0.90, 0.30]
```

The question is how could I merge (e.g., in a single list or dataframe the same parameter (i.e., Grouppre) from multiple "smXXX" objects (>300) in the environment. Ideally, something like:

```
Parameter        | Std. Coef. |    95% CI
----------------------------------------
Grouppre sm001   |     -0.30  | [-0.90, 0.30]
Grouppre sm002   |      0.42  | [-0.20, 0.60]
Grouppre sm101   |      0.51  | [-0.10, 0.80]
....
Grouppre sm301   |    - 0.18  | [-0.40, 0.10]
```

**Comment by JulianGaviriaL** [comment]
@mattansb 

Your option did not work:

```
mod_names <- ls() |> Filter(f = \(x) startsWith(x, "sm"))
> mod_list <- lapply(setNames(nm = mod_names), get)
> 
> parameters::compare_parameters(mod_list, 
+                                keep = "Grouppre", 
+                                standardize = "refit")
```
Following objects are not supported: sm001, sm002, sm003, sm004, sm005, sm006, sm007, sm008,
  sm009, sm010, sm011, sm012, sm013, sm014, sm015, sm016, sm017, sm018, sm019, sm020, sm021,
  sm022, sm023, sm024, sm025, sm026, sm027, sm028, sm029, sm030, sm031, sm032, sm033, sm034,
  sm035, sm036, sm037, sm038, sm039, sm040, sm041, sm042, sm043, sm044, sm045, sm046, sm047,
  sm048, sm049, sm050, sm051, sm052, sm053, sm054, sm055, sm056, sm057, sm058, sm059, sm060,
  sm061, sm062, sm063, sm064, sm065, sm066, sm067, sm068, sm069, sm070, sm071, sm072, sm073,
  sm074, sm075, sm076, sm077, sm078, sm079, sm080, sm081, sm082, sm083, sm084, sm085, sm086,
  sm087, sm088, sm089, sm090, sm091, sm092, sm093, sm094, sm095, sm096, sm097, sm098, sm099,
  sm100, sm101, sm102, sm103, sm104, sm105, sm106, sm107, sm108, sm109, sm110, sm111, sm112,
  sm113, sm114, sm115, sm116, sm117, sm118, sm119, sm120, sm121, sm122, sm123, sm124, sm125,
  sm126, sm127, sm128, sm129, sm130, sm131, sm132, sm133, sm134, sm135, sm136, sm137, sm138,
  sm139, sm140, sm141, sm142, sm143, sm144, sm145, sm146, sm147, sm148, sm149, sm150, sm151,
  sm152, sm153, sm154, sm155, sm156, sm157, sm158, sm159, sm160, sm161, sm162, sm163, sm164,
  sm165, sm166, sm167, sm168, sm169, sm170, sm171, sm172, sm173, sm174, sm175, sm176, sm177,
  sm178, sm179, sm180, sm181, sm182, sm183, sm184, sm185, sm186, sm187, sm188, sm189, sm190,
  sm191, sm192, sm193, sm194, sm195, sm196, sm197, sm198, sm199, sm200, sm201, sm202, sm203,
  sm204, sm205, sm206, sm207, sm208, sm209, sm210, sm211, sm212, sm213, sm214, sm215, sm216,
  sm217, sm218, sm219, sm220, sm221, sm222, sm223, sm224, sm225, sm226, sm227, sm228, sm229,
  sm230, sm232, sm233, sm234, sm235, sm236, sm237, sm238, sm239, sm240, sm241, sm242, sm243,
  sm244, sm245, sm246, sm247, sm248, sm249, sm250, sm251, sm252, sm253, sm254, sm255, sm256,
  sm257, sm258, sm259, sm260, sm261, sm262, sm263, sm264, sm265, sm266, sm267, sm268, sm269,
  sm270, sm272, sm273, sm274, sm275, sm276, sm277, sm278, sm279, sm280, sm281, sm282, sm283,
  sm284, sm285, sm286, sm287, sm288, sm289, sm290, sm291, sm292, sm293, sm294, sm295, sm296,
  sm297, sm298, sm299, sm300, sm301, sm302, sm303, sm304, sm305, sm306, sm307, sm308, sm309,
  sm310, sm311, sm312, sm313, sm314, sm315, sm316, sm317, sm318, sm319, sm320, sm321, sm322,
  sm323, sm324, sm325, sm326, sm327, sm328, sm329, sm330, sm331, sm332, sm333, sm334, sm335,
  sm336, sm337, sm338, sm339, sm340, sm341, sm342, sm343, sm344, sm345, sm346, sm347, sm348,
  sm349, sm350, sm351, sm352, sm353, sm354, sm355, sm356, sm357, sm358, sm359, sm360, sm361,
  sm362, sm363, sm364, sm365, sm366, sm367, sm368, sm369, sm370, sm371, sm372, sm373, sm374
  Dropping unsupported models now.
Error in startsWith(colnames(all_models), "model") : 
  non-character object(s)

**Reply by mattansb** [reply]
You need to call the models, not the standardized tables. See my example.

**Comment by JulianGaviriaL** [comment]
[standardized_pars.xlsx](https://github.com/easystats/parameters/files/13601650/standardized_pars.xlsx)
thanks @mattansb 
It worked!
As a suggestion... the wide /horizontal output might be problematic for multiple models. 
Thank you all for this nice package
